module.c

```diff
@@ -442,6 +509,77 @@ static PyObject *context_gc(ContextData *self) {
 	Py_RETURN_NONE;
 }
 
+static JSValue js_c_function(
+    JSContext *ctx, JSValueConst this_val, int argc, JSValueConst *argv, int magic) {
+	ContextData *self = (ContextData *)JS_GetContextOpaque(ctx);
+	PythonCallableNode *node = self->python_callables;
+	while (node && node->magic != magic) {
+		node = node->next;
+	}
+	if (!node) {
+		return JS_ThrowInternalError(self->context, "Internal error.");
+	}
+
+	PyObject *args = PyTuple_New(argc);
+	if (!args) {
+		return JS_ThrowOutOfMemory(self->context);
+	}
+	for (int i = 0; i < argc; ++i) {
+		PyObject *arg = quickjs_to_python(self, JS_DupValue(self->context, argv[i]));
+		PyTuple_SET_ITEM(args, i, arg);
+	}
+
+	PyObject *result = PyObject_CallObject(node->obj, args);
+	Py_DECREF(args);
+	if (!result) {
+		return JS_ThrowInternalError(self->context, "Python call failed.");
+	}
+	JSValue js_result = JS_NULL;
+	if (python_to_quickjs_possible(self, result)) {
+		js_result = python_to_quickjs(self, result);
+	}
+	Py_DECREF(result);
+	return js_result;
+}
+
+static PyObject *context_add_callable(ContextData *self, PyObject *args) {
+	const char *name;
+	PyObject *callable;
+	if (!PyArg_ParseTuple(args, "sO", &name, &callable)) {
+		return NULL;
+	}
+	if (!PyCallable_Check(callable)) {
+		PyErr_SetString(PyExc_TypeError, "Argument must be callable.");
+		return NULL;
+	}
+
+	PythonCallableNode *node = PyMem_Malloc(sizeof(PythonCallableNode));
+	if (!node) {
+		return NULL;
+	}
+	Py_INCREF(callable);
+	self->can_release_gil = 0;
+	node->magic = 0;
+	if (self->python_callables) {
+		node->magic = self->python_callables->magic + 1;
+	}
+	node->obj = callable;
+	node->next = self->python_callables;
+	self->python_callables = node;
+
+	JSValue function = JS_NewCFunctionMagic(
+	    self->context,
+	    js_c_function,
+	    "python",  // TODO: Name of the function should not matter that much?
```

Comment:
indeed it only affects `.name` in JS, which is mostly used for debugging. Should be fine to pass in the `name` variable though, quickjs atomizes so it doesn't have to be static afaict

Reply:
use `name` here, maybe it makes for nicer stack traces